packages/ListBox/src/components/Option/useIsSelectedOption.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import useListBox from "../../useListBox";
+import { selectSingleOption, selectMultipleOption } from "../Options/helpers/options";
+
+// previnting to report twice the onChange when props changed.
+const noop = () => {};
+
+export default function useIsSelectedOption({ index, props }) {
+  const [state, dispatch] = useListBox();
+
+  React.useEffect(() => {
+    if (props.isSelected !== null) {
+      if (state.isMulti) {
+        selectMultipleOption({
+          activeOptionIndex: index,
+          state,
+          dispatch,
+          isSelected: props.isSelected,
+          onChange: noop,
+        });
+        return;
+      }
+
+      if (props.isSelected === true) {
+        // we want to execute the method exclusively on the prop with selected === true
+        selectSingleOption({ activeOptionIndex: index, state, dispatch, isOpen: false, onChange: noop });
+      }
+    }
+  }, [dispatch, index, props.isSelected]); // eslint-disable-line
```

Comment:
I'm removing state from the deps if not this hook will keep firing all the time.